File: app/src/main/java/com/vicky7230/chat/data/network/XmppHelper.java
package com.vicky7230.chat.data.network;

import org.jivesoftware.smack.AbstractXMPPConnection;
import org.jivesoftware.smack.chat2.ChatManager;
import org.jivesoftware.smack.roster.Roster;
import org.jivesoftware.smack.roster.RosterEntry;

import java.util.Collection;

import io.reactivex.Observable;

/**
 * Created by vicky on 11/11/17.
 */

public interface XmppHelper {
    Observable<AbstractXMPPConnection> connect();

    Observable<AbstractXMPPConnection> login(String userName, String password);

    Observable<AbstractXMPPConnection> disconnect();

    ChatManager getChatManager();

    Roster getRoster();

    Observable<Boolean> sendMessage(String message);

    Observable<Collection<RosterEntry>> getRosterEntries();
}

File: app/src/main/java/com/vicky7230/chat/data/network/XmppHelper.kt
package com.vicky7230.chat.data.network

import io.reactivex.Observable
import org.jivesoftware.smack.AbstractXMPPConnection
import org.jivesoftware.smack.chat2.ChatManager
import org.jivesoftware.smack.roster.Roster
import org.jivesoftware.smack.roster.RosterEntry
import org.jivesoftware.smackx.search.UserSearchManager

/**
 * Created by vicky on 26/11/17.
 */
interface XmppHelper {

    fun connect(): Observable<AbstractXMPPConnection>

    fun login(userName: String, password: String): Observable<AbstractXMPPConnection>

    fun disconnect(): Observable<AbstractXMPPConnection>

    fun getChatManager(): ChatManager

    fun getRoster(): Roster

    fun getUserSearchManager(): UserSearchManager

    fun sendMessage(message: String): Observable<Boolean>

    fun getRosterEntries(): Observable<Collection<RosterEntry>>
}
